src/dynamics/prognostic_variables.jl

```diff
@@ -137,20 +135,35 @@ export PrognosticVariables
     clock::Clock = Clock()
 end
 
-get_nsteps(progn::PrognosticVariables{T, A, S, G, nsteps}) where {T, A, S, G, nsteps} = nsteps
 Base.eltype(progn::PrognosticVariables{T}) where T = T
 
+function get_steps(coeffs::LowerTriangularArray{T, 2}) where T
+    nsteps = size(coeffs, 2)
+    return ntuple(i -> lta_view(coeffs, :, i), nsteps)
+end
+
+function get_steps(coeffs::LowerTriangularArray{T, 3}) where T
+    nsteps = size(coeffs, 3)
+    return ntuple(i -> lta_view(coeffs, :, :, i), nsteps)
+end
+
+export get_step
+get_step(coeffs::LowerTriangularArray{T, 2}, i) where T = lta_view(coeffs, :, i)
+get_step(coeffs::LowerTriangularArray{T, 3}, i) where T = lta_view(coeffs, :, :, i)
+
+get_nsteps(progn::PrognosticVariables{T, A, S, G, nsteps}) where {T, A, S, G, nsteps} = nsteps
```

Comment:
does this work? nsteps isn't a parameter anymore? one could just do `=progn.nsteps` instead